File: 0766-toeplitz-matrix/0766-toeplitz-matrix.java
class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        for row in range(len(matrix)):
            for col in range(len(matrix[row])):
                if row>0 and col>0 and matrix[row][col]!=matrix[row-1][col-1] :
                    return False
        
        return True

File: 0766-toeplitz-matrix/0766-toeplitz-matrix.cpp
class Solution {
public:
    bool isToeplitzMatrix(vector<vector<int>>& matrix) {
        for(int row=0;row<matrix.size();++row)
            for(int col=0;col<matrix[row].size();++col)
                if(row>0&&col>0&&matrix[row][col]!=matrix[row-1][col-1])
                    return false;
        
        return true;
    }
};